controllers/virtualmachineservice/virtualmachineservice_controller.go

```diff
@@ -392,6 +393,11 @@ func (r *ReconcileVirtualMachineService) createOrUpdateService(ctx *context.Virt
 		service.Spec.ExternalName = vmService.Spec.ExternalName
 		service.Spec.LoadBalancerIP = vmService.Spec.LoadBalancerIP
 		service.Spec.LoadBalancerSourceRanges = vmService.Spec.LoadBalancerSourceRanges
+		if service.Spec.Type == corev1.ServiceTypeLoadBalancer {
+			service.Spec.AllocateLoadBalancerNodePorts = pointer.Bool(false)
+		} else {
+			service.Spec.AllocateLoadBalancerNodePorts = nil
```

Comment:
Do we need to specify the default behavior?

Reply:
To be explicit & the CreateOrUpdate() doesn't have an otherwise unexpected change during the steady-state reconcile.